### ISSUE TITLE
Support for Visual Studio 15

### ISSUE THREAD

**Issue by jurosh** [ReactWindows/react-native-windows#758]
Hi there,
I know maybe it's a little early (as VS 15 is still in Preview), but still I would like to ask if there is some support to run react-native with this new tool.
Or are there plans to add support for VS15 anytime soon ?
Cheers.

https://www.visualstudio.com/en-us/news/releasenotes/vs15-relnotes

**Comment by rozele** [comment]
@jurosh In theory there shouldn't be anything stopping you from opening the solution with VS 15, but I haven't tried it yet.

**Comment by jurosh** [comment]
I've tried it already, but with no luck.

Running of command `react-native run-windows` ended with NuGet error about `...CSharp.targets` - it's trying to search in directory `C:\Program Files (x86)\MSBuild\Microsoft\WindowsXaml\v14.0` but I found only `v15.0` on my disc (I suppose this is because of installed VS). I've tried to rename occurences of targets to some of new one in v15 but it doesnt helped.

In IDE it said that I should have VS14 (2015) and Severity	Code 
```
The package Microsoft.CSharp with version 4.0.1 could not be found in C:\Users\jurosh\.nuget\packages. Run a NuGet package restore to download the package.
```
It also saying something about retargeting - so this may be the way (but cannot find the mentioned button :) ), so I just changed chakra bridge version by some props:
```
Error	MSB8020	The build tools for v140 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install v140 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution".	ChakraBridge	D:\Programs\Visual Studio\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.target
```

After I upgraded it somehow it says:
```
Error	TRK0005	Failed to locate: "CL.exe". The system cannot find the file specified.	ChakraBridge	...
```

So it will not be totally straightforward :)

But I am more like React as VS/C# guy so if anyne with experience in development UWP may take a look and try.

**Comment by joewood** [comment]
I got this to work, these were my steps (this is with VS15 Preview 5 being the only install Visual Studio).
* ChakraBridge complains about the toolset version v140 not being available. I switched to v141 (VS2016) using Project Properties.
* Compiling against SDK 10.0.10240.0 returns build errors in ChakraBridge - `CompareFileTime`, `JsRunSerializedScriptWithCallback`, `JsGetTypedArrayInfo` not found. These are [10.0.10586.0 APIs only](https://msdn.microsoft.com/en-us/library/windows/apps/mt644399.aspx).
* Retarget to 10.0.10586 (min and max) fixes this - ChakraBridge builds OK
* Compiling ReactNative (C#) project now returns errors - `type forwarder for type 'Windows.UI.ViewManagement.StatusBar' causes a cycle` and `StatusBar` not found.
* Adding a reference to the Mobile Extension SDK [fixes this](http://stackoverflow.com/questions/32725185/using-statusbar-in-windows-universal-app). 

I'm not sure why the platform tools upgrade has caused SDK API resolution errors, but I think this excludes targeting 10240 (pre-Anniversary) versions of Windows 10.
